test/rematch2.jl

```diff
@@ -593,7 +607,7 @@ end
                 @test ex isa LoadError
                 e = ex.error
                 @test e isa ErrorException
-                @test e.msg == "$file:$line: Unrecognized @match2 block syntax: `b + c`."
+                @test e.msg == "$file:$line: Unrecognized @match block syntax: `b + c`."
```

Comment:
the error doesn't match the macro - is this intentional?

Reply:
Yes.  Soon there will only be one macro `@match`, and the code that produces the diagnostics will be shared with the "simple" implementation, which is currently called `@match` but will be renamed to `@match_simple` or something just for testing and diagnosing the implementation.